### PR TITLE
Continuation of UI for configuring storage proposal parameters

### PR DESCRIPTION
**Pull request by ancorgs**
This PR is a continuation of #397 adding some commits on top to change the behavior. For rationale see the comments in the original PR.

### How is this PR different from the original?

* It fully honors the specification at https://github.com/yast/yast-storage-ng/blob/master/doc/old_and_new_proposal.md treating all volumes equally, not enforcing different defaults or a different UI for root/swap/home/rest. Thus, it retains all the flexibility of the control file format and allows all extra scenarios... the useful ones and the stupid ones.

* It respects the volumes order coming from the control file.

* It does not show volumes that the user cannot configure, and even skips the full step if the user can't configure any setting.
  * That's more consistent with the current design of the guided setup, that only shows widgets and steps for those things that can be actually tweaked.
  * That's more consistent in general for cases like swap (that was sometimes shown and sometimes not in the previous version) or partitions needed for booting.
  * The right place for the full list is the proposal screen, after all.

* It produces more specific string for the different volume types and for LVM vs partitions, to allow more flexibility in the translations. For example the former `_("Propose Separate %{name} Partition")`  (with "name" being Root/Swap/Home or a mount point) didn't fit very well the Spanish case that would be completely different for Swap (something like "._..una partición de intercambio aparte_") and for a mount point (something like "_...una partición aparte para %{name}_").

Screenshot showing two mandatory but configurable partitions, and two optional ones. In a arbitrary order decided by the release manager. The implementation at #397 made impossible, for example, to have such order or to have an optional swap partition.

new_control

Screenshot showing basically the same screen but having selected LVM in the previous screen. Note the slightly different strings ("Volume" instead of "Partition").

new_control_lvm

### CODE REVIEW

File: test/y2storage/dialogs/guided_setup/select_filesystem_ng_test.rb [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+
+    context "if the user can decide whether to enlarge by ram" do
+      let(:index) { 4 }
+
+      it "includes the corresponding check box" do
+        expect(adjust_by_ram_checkbox.value).to eq :CheckBox
+      end
+    end
+
+    context "if the user can not decide whether to enlarge by ram" do
```
Comment (joseivanlopez) [inline test/y2storage/dialogs/guided_setup/select_filesystem_ng_test.rb]: Typo: can not -> cannot

File: src/lib/y2storage/dialogs/guided_setup/base.rb [hunk 2]
```diff
@@ -95,6 +95,7 @@ def create_dialog
 
         # Should be implemented by derived classes.
         def initialize_widgets
+          log.error("ERROR: initialize_widgets not implemented by derived class")
           nil
```
[the remark this answers is not in the document]
Comment (ancorgs) [reply src/lib/y2storage/dialogs/guided_setup/base.rb]: In fact, it's not even worth a warning. It's perfectly fine that a dialog don't need to perform `initialize_widgets`, which is there just to support some specific cases (enabling and disabling widgets right after drawing the dialog).

File: src/lib/y2storage/volume_specification.rb [hunk 3]
```diff
@@ -171,23 +214,19 @@ def load_features(volume_features)
       apply_fallbacks
     end
 
+    # @param types [Array<String>, String] an array of filesystem types or a
+    #   list of comma-separated ones
     def fs_types=(types)
+      types = types.strip.split(/\s*,\s*/) if types.is_a?(String)
```
[the remark this answers is not in the document]
Comment (ancorgs) [reply src/lib/y2storage/volume_specification.rb]: The current one works better when there are spaces around the comma, like "btrfs, ext2" or even "btrfs , ext2". With your version you would need an extra strip in every element of the result.

[the remark this answers is not in the document]
Comment (joseivanlopez) [reply src/lib/y2storage/volume_specification.rb]: Oh, right!

**Pull request merged**